src/aggregator/aggregation/timer.go

```diff
@@ -81,6 +71,17 @@ func (t *Timer) AddBatch(timestamp time.Time, values []float64) {
 	}
 
 	t.stream.AddBatch(values)
+
+	// Keep the last annotation which was set.
+	if len(annotation) > 0 {
+		if cap(t.annotation) < len(annotation) {
+			// Twice as long in case another one comes in
+			// and we could avoid realloc as long as less than this first alloc.
+			t.annotation = make([]byte, 0, 2*len(annotation))
+		}
+		// Reuse any previous allocation while taking a copy.
+		t.annotation = append(t.annotation[:0], annotation...)
+	}
```

Comment:
just make a helper method, this seems to be sprinkled all over the codebase.
are you sure that the reallocation sizing is even worth it? might make memory usage higher than really needed.